psydac/linalg/tests/test_iterative_solvers.py

```diff
@@ -0,0 +1,498 @@
+import numpy as np
+import pytest
+from psydac.linalg.iterative_solvers import bicg, bicgstab, lsmr, minres, pcg, jacobi, cg
+from psydac.linalg.stencil import StencilVectorSpace, StencilMatrix, StencilVector
+from psydac.linalg.basic import LinearSolver
+from psydac.ddm.cart import DomainDecomposition, CartDecomposition
+def define_data_hermitien(n, p, dtype=float):
+    domain_decomposition = DomainDecomposition([n - p], [False])
+    cart = CartDecomposition(domain_decomposition, [n], [np.array([0])], [np.array([n - 1])], [p], [1])
+    # ... Vector Spaces
+    V = StencilVectorSpace(cart,dtype=dtype)
+    e = V.ends[0]
+    s = V.starts[0]
+
+    # Build banded matrix with 2p+1 diagonals: must be symmetric and positive definite
+    # Here we assign value 2*p on main diagonal and -1 on other diagonals
+    if dtype==complex:
+        factor=1+1j
+    else:
+        factor=1
+    A = StencilMatrix(V, V)
+    A[:, -p:0] = 1-1*factor
+    A[:, 0:1] = 2 * p
+    A[:, 1:p + 1] = 1-1*factor.conjugate()
+    A.remove_spurious_entries()
+
+    # Build exact solution
+    xe = StencilVector(V)
+    xe[s:e + 1] = factor*np.random.random(e + 1 - s)
+    return(V, A, xe)
+
+def define_data(n, p, matrix_data, dtype):
+    domain_decomposition = DomainDecomposition([n - p], [False])
+    cart = CartDecomposition(domain_decomposition, [n], [np.array([0])], [np.array([n - 1])], [p], [1])
+    # ... Vector Spaces
+    V = StencilVectorSpace(cart, dtype=dtype)
+    e = V.ends[0]
+    s = V.starts[0]
+
+    # Build banded matrix with 2p+1 diagonals: must be symmetric and positive definite
+    # Here we assign value 2*p on main diagonal and -1 on other diagonals
+
+    A = StencilMatrix(V, V)
+    A[:, -p:0] = -matrix_data[0]
+    A[:, 0:1] = matrix_data[1]
+    A[:, 1:p + 1] = matrix_data[2]
+    A.remove_spurious_entries()
+
+    # Build exact solution
+    xe = StencilVector(V)
+    xe[s:e + 1] = np.random.random(e + 1 - s)
+    return(V, A, xe)
+
+#===============================================================================
+@pytest.mark.parametrize( 'n', [5, 10, 13] )
+@pytest.mark.parametrize('p', [2, 3])
+@pytest.mark.parametrize('dtype', [float,complex])
+def test_bicgstab_tridiagonal( n, p, dtype ):
+    """
+    Test generic Biconjugate Gradient algorithm on tridiagonal linear system.
+
+    Parameters
+    ----------
+    n : int
+        Dimension of linear system (number of rows = number of columns).
```

Comment:
If you really want to include detailed docstrings in the unit tests (usually not necessary) then you need to update the `Parameters` section.

Reply:
[86b06e9](https://github.com/pyccel/psydac/pull/273/commits/86b06e97fda7a028f2d6ec3f23b6c81d44eaf696)